packages/styletron-react/src/provider.js

```diff
@@ -1,44 +1,18 @@
 const React = require('react');
+const createProvider = require('./create-provider.js');
 
-/**
- * @class StyletronProvider
- * @packagename styletron-react
- * @description React component
- * @example
- * const Styletron = require('styletron');
- *
- * function render() {
- *   return React.renderToString(
- *     <StyletronProvider styletron={new Styletron()}>
- *       <App/>
- *     </StyletronProvider>
- *   );
- * }
- * 
- * @property {object} styletron - Styletron instance
- * @property {ReactElement} children - children
- * @extends ReactClass
- */
-class StyletronProvider extends React.Component {
-  getChildContext() {
-    return {styletron: this.styletron};
+module.exports = createProvider(
+  ({styletron, injectStyle}) => {
+    if (!injectStyle && styletron) {
+      throw new Error('You need to switch to the exported `StyletronProvider` from `styletron-react/featured` in this upgraded version. For more info please visit: https://github.com/rtsao/styletron/');
+    }
+    return {
+      styletron,
+      injectStyle
```

Comment:
Realized that other libs expects the instance to be on `context.styletron`. Need to find a better way to pass `injectStyle` without adding a new property to `context`.